include/envoy/server/filter_config.h

```diff
@@ -188,19 +208,28 @@ class NamedListenerFilterConfigFactory {
   virtual Network::ListenerFilterFactoryCb
   createFilterFactoryFromProto(const Protobuf::Message& config,
                                ListenerFactoryContext& context) PURE;
+};
 
-  /**
-   * @return ProtobufTypes::MessagePtr create empty config proto message for v2. The filter
-   *         config, which arrives in an opaque message, will be parsed into this empty proto.
-   *         Optional today, will be compulsory when v1 is deprecated.
-   */
-  virtual ProtobufTypes::MessagePtr createEmptyConfigProto() PURE;
+/**
+ * Implemented by each udp listener filter and registered via Registry::registerFactory()
+ * or the convenience class RegisterFactory.
+ */
+class NamedUdpListenerFilterConfigFactory : public ListenerFilterConfigFactoryBase {
+public:
+  virtual ~NamedUdpListenerFilterConfigFactory() {}
 
   /**
-   * @return std::string the identifying name for a particular implementation of a listener filter
-   * produced by the factory.
+   * Create a particular udp listener filter factory implementation. If the implementation is unable
+   * to produce a factory with the provided parameters, it should throw an EnvoyException in the
+   * case of general error or a Json::Exception if the json configuration is erroneous. The returned
```

Comment:
nit: no JSON, please audit comments for "v1isms"

Reply:
will do